app-backend/src/main/kotlin/org/example/appbackend/config/JwtTokenFilter.kt

```diff
@@ -0,0 +1,103 @@
+package org.example.appbackend.config
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.context.SecurityContextHolder
+import org.example.appbackend.service.UserService
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource
+import org.springframework.web.filter.OncePerRequestFilter
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.security.Keys
+import jakarta.servlet.FilterChain
+import jakarta.servlet.ServletException
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import java.io.IOException
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.Jws
+import io.jsonwebtoken.JwtException
+import java.security.Key
+import java.util.Date
+import java.util.*
+import org.slf4j.LoggerFactory
+
+class JwtTokenFilter(
+        @Value("\${spring.jwt.secret}")
+        private val jwtSecret: String,
+        private val userService: UserService
+) : OncePerRequestFilter() {
+
+    private var logger = LoggerFactory.getLogger(JwtTokenFilter::class.java)
+
+    @Throws(ServletException::class, IOException::class)
+    override fun doFilterInternal(
+            request: HttpServletRequest,
+            response: HttpServletResponse,
+            filterChain: FilterChain
```

Comment:
I guess the spacing there is not according to the official kotlin codestyle (I guess it should be 4 spaces, not 8)